### PR TITLE
Replace socketio with nats.ws

### PR DESCRIPTION
**Pull request by OrMemphis**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
             "mode": "auto",
             "program": "${workspaceFolder}/main.go",
             "args": [
-                "--js", "--auth", "memphis", "-D"
+                "--js", "--auth", "memphis", "-D", "-c", "memphis_nats.conf"
```
Comment (idanasulinmemphis) [inline .vscode/launch.json:11]: change to ws.conf

File: memphis_nats.conf [hunk 2]
```diff
@@ -0,0 +1,5 @@
+websocket
```
Comment (idanasulinmemphis) [inline memphis_nats.conf:1]: change this file name to ws.conf and locate it inside the conf directory

File: memphis_nats.conf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+websocket 
+{
+     port: 8080
```
Comment (idanasulinmemphis) [inline memphis_nats.conf:3]: change to 6667

File: server/memphis_handlers_ws.go [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+	switch subjectHead {
+	case memphisWS_Subj_MainOverviewData:
+		newSub.updateFunc = func() any {
+			result, err := memphisWSGetMainOverviewData(h)
+			return memphisWSExtractErr(s, result, err)
+		}
+	case memphisWS_Subj_StationOverviewData:
+		stationName := tokenAt(subj, 2)
+		if stationName == _EMPTY_ {
+			s.Errorf("memphis websocket: invalid station name")
```
Comment (idanasulinmemphis) [inline server/memphis_handlers_ws.go:146]: missing return statement

File: server/memphis_handlers_ws.go [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+	return memphisWSSubscription{
+		refCount: 1,
+	}
+}
+
+func memphisWSExtractErr(s *Server, result any, err error) any {
+	if err != nil {
+		s.Errorf(err.Error())
+	}
+	return result
```
Comment (idanasulinmemphis) [inline server/memphis_handlers_ws.go:193]: add else block

File: server/memphis_handlers_ws.go [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+		stationName := tokenAt(subj, 2)
+		if stationName == _EMPTY_ {
+			s.Errorf("memphis websocket: invalid station name")
+		}
+		newSub.updateFunc = func() any {
+			result, err := memphisWSGetStationOverviewData(s, h, stationName)
+			return memphisWSExtractErr(s, result, err)
+		}
+	case memphisWS_Subj_PoisonMsgJourneyData:
+		poisonMsgId := tokenAt(subj, 2)
```
Comment (idanasulinmemphis) [inline server/memphis_handlers_ws.go:153]: check whether not nil

File: ui_src/src/App.js [hunk 7]
```diff
@@ -80,17 +81,19 @@ const App = withRouter(() => {
...
-                        reconnection: false
-                    });
-                    dispatch({ type: 'SET_SOCKET_DETAILS', payload: socket });
+                    const conn = await connect(
+                        {
+                            servers: ["ws://localhost:8080"],
+                            token: "memphis",
+                        }
+                    );
+                    console.log(`connected to ${conn.getServer()}`);
```
Comment (avrhamNeeman) [inline ui_src/src/App.js:94]: Remove

**Review by idanasulinmemphis — APPROVED** [review]

**Pull request merged**